dig.go

```diff
@@ -60,24 +60,18 @@ type InvokeOption interface {
 
 // Container is a directed acyclic graph of types and their dependencies.
 type Container struct {
-	nodes map[key]*node
-	cache map[key]reflect.Value
-
-	// TODO: for advanced use-case, add an index
-	// This will allow retrieval of a single type, without specifying the exact
-	// tag, provided there is only one object of that given type
-	//
-	// It will also allow library owners to create a "default" tag for their
-	// object, in case users want to provide another type with a different name
-	//
-	// index map[reflect.Type]key
+	// Constructors for different values.
+	producers map[key][]*node
```

Comment:
I find it odd that something that stores `constructors` (godoc above), and keyed via `key -> *node` is called `producers`.

Reply:
The reason I called them producers instead of constructors is that constructor
almost implies one constructor per value type, but with this new structure we
can have multiple constructors which produce the same type of value.

So it made more sense to me to call them producers since they produce values.
I'll update the docs to reflect this a bit more.